mriqc/workflows/diffusion/base.py

```diff
@@ -416,6 +412,81 @@ def dmri_bmsk_workflow(name="dmri_brainmask", omp_nthreads=None):
     return workflow
 
 
+def init_dmriref_wf(
+    in_file=None,
+    multiecho=False,
+    name="init_dmriref_wf",
+):
+    """
+    Build a workflow that generates reference images for a dMRI series.
+
+    The raw reference image is the target of :abbr:`HMC (head motion correction)`, and a
+    contrast-enhanced reference is the subject of distortion correction, as well as
+    boundary-based registration to T1w and template spaces.
+
+    This workflow assumes only one dMRI file has been passed.
+
+    Workflow Graph
+        .. workflow::
+            :graph2use: orig
+            :simple_form: yes
+
+            from mriqc.workflows.diffusion.base import init_dmriref_wf
+            wf = init_dmriref_wf()
+
+    Parameters
+    ----------
+    in_file : :obj:`str`
+        dMRI series NIfTI file
+    ------
+    in_file : str
+        series NIfTI file
+
+    Outputs
+    -------
+    in_file : str
+        Validated DWI series NIfTI file
+    ref_file : str
+        Reference image to which DWI series is motion corrected
+    """
+    from niworkflows.interfaces.bold import NonsteadyStatesDetector
```

Comment:
Non-steady state detection was removed. @oesteban @arokem Is it common for there to be dummy scans at the beginning of a DWI series, or would it be a waste of time for diffusion?

Reply:
Removing definitely was a good call.

In DWI, some people calculate the S0 signal drift along the timeseries instead of detecting nonsteady states. Only some people do this because DWI models are generally voxelwise and do not use information aggregated over directions, so they are intrinsically robust against these drifts.

However, MRIQC does compute the drift to report it (and dMRIPrep will too). That's the point of the [`CorrectSignalDrift`](https://github.com/nipreps/mriqc/blob/83a66e0578420b1a086595647ed492897d59eb27/mriqc/interfaces/diffusion.py#L260) interface.

So the algorithm, in summary is:

1. Extract low-b volumes (or b=0, but normally you call this anything with b < threshold, with threshold typically between 50 and ~500).
2. Calculate "global signal" (expectation, use median for unbiased estimate) within brain mask.
3. Fit an exponential decay to the global signal within brain mask (which models the drift)
4. The BOLD nonsteady states are at the beginning of this exponential, when the signal most rapidly decays. DWI data generally acquire just a few (or one) low b at the beginning, and it is generally recommended to collate some other low-bs along evenly along the scan to have better sampling of the exponential drift.

I hope that explains this. Two thoughts:
- The original epi_reference workflow tried to allow for this kind of approach.
- Relatedly, we could attempt to implement this "drift" correction within fMRIPrep, or at least, generate a regressor that you can put replacing the intercept in the design matrix.

Reply:
Since this is very much related to the final classes of the last nipraxis edition, I wonder what @matthew-brett would think about the very last bullet point. I'm sure some people have included the exponential decay model as a regressor of the BOLD signal.

Also interested in @poldrack's opinion here :)

Reply:
I agree with removing the NSSD for DWI data.  I don't think I like the idea of using an exponential model to regress out the initial non-steady-state signal since it's not just an amplitude issue (at least that's my understanding), the contrast is also different which is why we generally drop those acquisitions altogether.